apps/desktop/desktop/src/main/java/bisq/desktop/components/overlay/TextInputDialog.java

```diff
@@ -0,0 +1,210 @@
+
+    public TextInputDialog(String headline) {
+        this(headline, null, null);
+    }
+
+    public TextInputDialog(String headline, String fieldLabel, String prompt) {
+        if (headline != null) {
+            headline(headline);
+        }
+
+        hideCloseButton();
+
+        GridPane gridPane = new GridPane();
+        gridPane.setHgap(10);
+        gridPane.setVgap(10);
+        gridPane.setPadding(new Insets(15, 20, 10, 14));
+        gridPane.setPrefWidth(PREFERRED_WIDTH);
+
+        ColumnConstraints columnConstraints = new ColumnConstraints();
+        columnConstraints.setHgrow(Priority.ALWAYS);
+        columnConstraints.setFillWidth(true);
+        columnConstraints.setPercentWidth(100);
+        gridPane.getColumnConstraints().add(columnConstraints);
+
+        instructionLabel = new Label();
+        instructionLabel.setWrapText(true);
+        instructionLabel.setManaged(false);
+        instructionLabel.setVisible(false);
+        instructionLabel.setMaxWidth(Double.MAX_VALUE);
+
+        textField = new MaterialTextField(fieldLabel, prompt);
+        textField.setMaxWidth(Double.MAX_VALUE);
+
+        counterLabel = new Label(Res.get("component.TextInputDialog.text.characters.remaining", maxLength));
+        counterLabel.setManaged(showCounter);
+        counterLabel.setVisible(showCounter);
+
+        textField.textProperty().addListener((observable, oldValue, newValue) -> {
+            if (newValue != null && newValue.length() > maxLength) {
+                textField.setText(oldValue);
+                return;
+            }
+            updateRemainingCharsCount();
+        });
+
+        confirmButton = new Button(confirmButtonText);
+        cancelButton = new Button(cancelButtonText);
+
+        confirmButton.getStyleClass().add("action-button");
+        confirmButton.setDefaultButton(true);
+
+        HBox buttonBox = new HBox(10, confirmButton, cancelButton);
+        buttonBox.setPadding(new Insets(10, 0, 0, 0));
+        buttonBox.setAlignment(Pos.CENTER_RIGHT);
+
+        int rowIndex = 0;
+        gridPane.add(instructionLabel, 0, rowIndex++);
+        gridPane.add(textField, 0, rowIndex++);
+        gridPane.add(counterLabel, 0, rowIndex++);
+        gridPane.add(buttonBox, 0, rowIndex);
+
+        confirmButton.setOnAction(e -> {
+            String input = textField.getText();
+            if (validator != null && !validator.test(input)) {
+                return;
+            }
+
+            if (inputConsumer != null) {
+                inputConsumer.accept(input);
+            }
+            hide();
+        });
+
+        cancelButton.setOnAction(e -> hide());
+
+        GridPane.setHgrow(textField, Priority.ALWAYS);
+        GridPane.setHgrow(buttonBox, Priority.ALWAYS);
+        GridPane.setHgrow(instructionLabel, Priority.ALWAYS);
+        GridPane.setHgrow(counterLabel, Priority.ALWAYS);
+
+        content(gridPane);
+    }
+
+    public TextInputDialog fieldLabel(String label) {
```

Comment:
Better stick with the existing terminology - `description` and stick to `set*` for setters

-> `setInputFieldDescription(String value)`